<launch>
    <!-- <node pkg="docking_tracker" name="path_tracker" type="nmpc_ros_with_origin.py">
        <param name="global_frame_id"           value="odom"/>
        <param name="local_frame_id"           value="base_footprint"/>
        <param name="path_topic_name"           value="/desired_path"/>
    </node> -->

    <node pkg="docking_tracker" name="nmpc_ros" type="nmpc_ros">
        <param name="control_frequency"         value="10.0"/>
        <param name="mpc_steps"                 value="20.0"/>
        <param name="ref_cte"                   value="0.1"/>
        <param name="ref_etheta"                value="0.1"/>
        <param name="ref_vel"                   value="0.3"/>
        <param name="w_cte"                     value="3000"/>
        <param name="w_etheta"                  value="200"/>
        <param name="w_vel"                     value="1000"/>
        <param name="w_angvel"                  value="10"/>
        <param name="w_accel"                   value="200"/>
        <param name="w_angvel_d"                value="10"/>
        <param name="w_accel_d"                 value="0.01"/>
        <param name="max_angvel"                value="0.2"/>
        <param name="max_throttle"              value="0.2"/>
        <param name="bound_value"               value="20.0"/>
        <param name="path_length"               value="5.0"/>
        <param name="max_linear_speed"          value="0.3"/>
        <param name="min_linear_speed"          value="-0.3"/>
        <param name="xy_tolerance"              value="0.02"/>
        <param name="yaw_tolerance"             value="0.02"/>
        
        <!-- <remap from="/cmd_vel"                  to="/cmdd_vel"/> -->
    </node>
</launch>